api/src/utils.ts

```diff
@@ -33,16 +33,16 @@ async function generateDomainsFile(): Promise<void> {
   return promisify(fs.writeFile)(path.join(config.domains_dir, config.domains_file), output);
 }
 
-function promisifyChildProcess(child: ChildProcess): Promise<void> {
+function promisifyChildProcess(child: ChildProcess, successMessage: string = null, errorMessage: string = null): Promise<void> {
   return new Promise((resolve, reject) => {
     
     child.addListener("error", reject);
     child.addListener("exit", (code) => {
-      console.log(`Child process exited with code ${code}`);
+      console.log(successMessage || `Child process exited with code ${code}`);
       if (code === 0) {
         resolve();
       } else {
-        reject("Child process exited with non-zero code");
+        reject(errorMessage || "Child process exited with non-zero code");
```

Comment:
This will still swallow the error. Error should always propagate up even if it's not pretty for user. Otherwise it's a nightmare to do support.

The entire function could become

```ts
return promisify(exec)(cmd)
```

Reply:
You're right, I solved this in the other PR